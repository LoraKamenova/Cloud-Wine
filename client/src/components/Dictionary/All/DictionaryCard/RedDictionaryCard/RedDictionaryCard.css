.red-dictionary-card {
    background-color: firebrick;
}

.red-dictionary-card-content-wrapper {
    color: white;
}

.red-dictionary-card-button {
    border: none;
    color: firebrick;
    background-color: white;
}

.red-dictionary-card-button:hover {
    background-color: gray;
}
