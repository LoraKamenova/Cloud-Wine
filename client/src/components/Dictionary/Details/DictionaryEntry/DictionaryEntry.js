import './DictionaryEntry.css'

const DictionaryEntry = (props) => {
    return (
        <article className="dictionary-article">
            <div className="dictionary-article-header" style={{backgroundImage: `url(${props.imageUrl})`}}>
                <span className="dictionary-article-heading">{props.title}</span>
            </div>
            <p className="dictionary-article-paragraph">{props.fullText}</p>
        </article>
    );
};

export default DictionaryEntry;
