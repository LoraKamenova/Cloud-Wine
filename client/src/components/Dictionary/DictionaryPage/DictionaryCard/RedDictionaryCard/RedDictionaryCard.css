.red-dictionary-card {
    display: flex;
    flex-direction: column;
    width: 200px;
    height: 300px;
    padding: 20px;
    box-shadow: 0px 0px 11px -4px #000000;
    justify-content: space-between;
    align-items: center;
    margin: 30px;
    background-color: firebrick;
}

.red-dictionary-card-content-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    color: white;
}

.red-dictionary-image {
    width: 60px;
    height: 60px;
}

.red-dictionary-card-title {
    text-transform: uppercase;
    letter-spacing: 2px;
    margin-bottom: 15px;
}

.red-dictionary-card-text {
    text-align: justify;
    font-size: 14px;
    margin-bottom: 10px;
}

.red-dictionary-card-button {
    border-radius: 30px;
    border: none;
    outline: none;
    padding: 7px 30px;
    text-decoration: none;
    text-transform: uppercase;
    letter-spacing: 1px;
    color: firebrick;
    background-color: white;
    font-weight: bold;
}

.red-dictionary-card-button:hover {
    cursor: pointer;
    color: white;
    background-color: gray;
}
