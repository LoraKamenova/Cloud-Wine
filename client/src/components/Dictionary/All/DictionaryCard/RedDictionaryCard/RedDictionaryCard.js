import './RedDictionaryCard.css'
import {Link} from "react-router-dom";

const RedDictionaryCard = (props) => {
    return (
        <article className="red-dictionary-card">
            <div className="red-dictionary-card-content-wrapper">
                <div className="red-dictionary-image-wrapper">
                    <img className="red-dictionary-image" src={props.iconUrl} alt="icon"></img>
                </div>
                <h3 className="red-dictionary-card-title">{props.title}</h3>
                <div className="red-dictionary-text-wrapper">
                    <p className="red-dictionary-card-text">{props.shortText}</p>
                </div>
                <Link className="red-dictionary-card-button" to={props.link}>Прочети</Link>
            </div>
        </article>
    );
};

export default RedDictionaryCard;
