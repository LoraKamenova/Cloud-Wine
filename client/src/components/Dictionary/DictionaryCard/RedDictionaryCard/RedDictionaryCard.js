import './RedDictionaryCard.css'
import {Link} from "react-router-dom";

const RedDictionaryCard = (props) => {
    return (
        <article className="red-dictionary-card">
            <div className="red-dictionary-card-content-wrapper">
                <div className="red-dictionary-image-wrapper">
                    <img className="red-dictionary-image" src={props.url}></img>
                </div>
                <h3 className="red-dictionary-card-title">{props.title}</h3>
                <p className="red-dictionary-card-text">{props.text}</p>
            </div>
            <Link className="red-dictionary-card-button" to={props.link}>Прочети</Link>
        </article>
    );
};

export default RedDictionaryCard;
