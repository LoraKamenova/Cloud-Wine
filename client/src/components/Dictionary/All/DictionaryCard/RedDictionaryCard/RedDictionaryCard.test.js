import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import RedDictionaryCard from './RedDictionaryCard';
import DictionaryEntry from "../../../Details/DictionaryEntry/DictionaryEntry";

describe('RedDictionaryCard Component', () => {
    it('Should display entry title', () => {
        render(
            <BrowserRouter>
                <RedDictionaryCard title="Аериране" link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".red-dictionary-card-title").textContent).toBe('Аериране');
    });

    it('Should display card content', () => {
        render(
            <BrowserRouter>
                <RedDictionaryCard shortText="Аерирането на виното означава излагане на виното на въздух или..." link="/"/>
            </BrowserRouter>
        );

        expect(document.querySelector(".red-dictionary-card-text").textContent).toBe('Аерирането на виното означава излагане на виното на въздух или...');
    });

    it("RedDictionaryCard Component snapshot", () => {
        const { asFragment } = render(<BrowserRouter><RedDictionaryCard link="/"/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

});
