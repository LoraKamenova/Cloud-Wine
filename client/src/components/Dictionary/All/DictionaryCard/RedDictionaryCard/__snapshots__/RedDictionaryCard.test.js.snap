// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedDictionaryCard Component RedDictionaryCard Component snapshot 1`] = `
<DocumentFragment>
  <article
    class="red-dictionary-card"
  >
    <div
      class="red-dictionary-card-content-wrapper"
    >
      <div
        class="red-dictionary-image-wrapper"
      >
        <img
          alt="icon"
          class="red-dictionary-image"
        />
      </div>
      <h3
        class="red-dictionary-card-title"
      />
      <div
        class="red-dictionary-text-wrapper"
      >
        <p
          class="red-dictionary-card-text"
        />
      </div>
      <a
        class="red-dictionary-card-button"
        href="/"
      >
        Прочети
      </a>
    </div>
  </article>
</DocumentFragment>
`;
